tests/devices/test_default_qubit.py

```diff
@@ -1733,3 +1734,15 @@ def test_wires_expval(self, wires1, wires2, tol):
         circuit2 = self.make_circuit_expval(wires2)
 
         assert np.allclose(circuit1(), circuit2(), tol)
+
+
+def test_get_slice():
+    """Test that the _get_slice function returns the expected slice and allows us to slice
+    correctly into an array."""
+
+    sl = _get_slice(1, 1, 3)
+    array = np.arange(27).reshape((3, 3, 3))
+    target = array[:, 1, :]
+
+    assert sl == (slice(None, None, None), 1, slice(None, None, None))
+    assert np.allclose(array[sl], target)
```

Comment:
Nice! Is it worth testing this for additional edge cases? E.g., 1D arrays, arrays with dimension of size 1, e.g., ``[[0, 1]]`